### PR TITLE
Typescript node discriminator fixes

### PR DESCRIPTION
**Pull request by bjarnij**
### PR checklist

- [X] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [X] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [X] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [X] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR
The generated code for discriminators that are not `undefined discriminators`, lacked quotes around the name of the property containing the discriminator causing the generated code not to compile.
Also I explicitly defined the undefined discriminator as any to get rid of the `[ts] Member 'discriminator' implicitly has an 'any' type.` error message from the TypeScript compiler when the compiler option `noImplicitAny` is set to true.

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny

### CODE REVIEW

File: modules/swagger-codegen/src/main/resources/typescript-node/api.mustache [hunk 1]
```diff
@@ -145,10 +145,10 @@ export class {{classname}} {{#parent}}extends {{{parent}}} {{/parent}}{
 {{/vars}}
 
     {{#discriminator}}
-    static discriminator = {{discriminator}};
+    static discriminator = '{{discriminator}}';
     {{/discriminator}}
     {{^discriminator}}
-    static discriminator = undefined;
+    static discriminator : any = undefined;
```
[the remark this answers is not in the document]
Comment (bjarnij) [reply modules/swagger-codegen/src/main/resources/typescript-node/api.mustache]: @TiFu Yes type `string` is correct. I've updated the PR to according to your comments.

**Comment by gbrown-ce** [comment]
@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx @macjohnny @bjarnij

Any chance we can get this to merge into 2.4? Its blocking our ability to use discriminators in our current workflow.

**Comment by gbrown-ce** [comment]
@wing328 I think you can close this PR now after merging this: https://github.com/swagger-api/swagger-codegen/pull/8157

**Comment by gbrown-ce** [comment]
@wing328 do you know who is responsible for updating this? https://hub.docker.com/r/swaggerapi/swagger-codegen-cli/

I'd love to get this latest changed into that image.

**Comment by wing328** [comment]
@gbrown-ce I'll look into that...

**Comment by gbrown-ce** [comment]
@wing328 you're awesome, thank you!

**Pull request closed without merging**